hedera-node/docs/design/services/smart-contract-service/accounts-blocklisting.md

```diff
@@ -0,0 +1,68 @@
+- In Ethereum, everyone that knows the corresponding private key will be able to authorize a transaction from the known address, and the first to send a transaction would be eligible to withdraw the funds.
+- In Hedera, only the account with the associated corresponding alias would be eligible to withdraw the funds. If another key controls this account, then only that key can be used to withdraw the funds.
+
+For more information about EVM addresses, aliases and how they are used in Hedera see [Auto Account Creation](https://docs.hedera.com/hedera/core-concepts/accounts/auto-account-creation).
+
+### Example user story
+
+As a dApp developer, I have written some contracts and tests for these contracts.
+
+In Ethereum:
+- While executing tests against different environments, using some default private test keys (e.g. the ones configured with Hardhat) to create and interact with accounts, I accidentally executed my tests on Mainnet and transferred some funds to an account with public address derived from one of the default test keys.
+- As those default private test keys are well-known, anyone can use them to withdraw the funds from the account, so in effect, the funds are lost.
+
+In Hedera (current behavior):
+- While executing tests against different environments, using some default private test keys (e.g. the ones configured with Hedera Local Node) to create and interact with accounts, I accidentally executed my tests on Mainnet and transferred some funds to an account with alias derived from one of the Hedera Local Node keys.
+- As those default private test keys are well-known, anyone could have used them to create an account with the same alias but with different account admin key, leaving me with no control over the funds.
+
+In Hedera (desired behavior):
+- While executing tests against different environments, using some default private test keys (e.g. the ones configured with Hedera Local Node) to create and interact with accounts, I accidentally executed my tests on Mainnet and transferred some funds to an account with alias derived from one of the Hedera Local Node keys.
+- As those default private test keys are well-known, the ledger can be configured to own those EVM addresses and block any transfers to them, so that I cannot accidentally send funds to them.
+
+## Goals
+
+- We want to block a list of known addresses, so that no one can “squat” on them and exercise full control on any accidentally sent and blocked funds. Sending funds to such blocked addresses should not be successful as these accounts are considered “compromised”.
+
+## Non Goals
+
+- We don't consider unblocking already blocked accounts.
+- We don't consider what happens to a blocked account when it expires.
+
+## Architecture
+
+The implementation related to blocked accounts will be gated by a feature flag.
+
+### File containing the list of blocked accounts
+We will store the list of blocked accounts in a regular CSV resource file.
```

Comment:
With this approach we can update the blocked accounts list at every release (if/as necessary).  Is it possible we may want to update it more often, or just between releases?  In that case a proper Hedera file could be used?  (I believe that's used for other properties already.)

(If this has been considered and rejected possibly a sentence could be added to the doc about "alternative approaches considered" - I've seen that in other design docs.)

Reply:
Hmm, we may consider this. What are the advantages of using a "proper Hedera file"? 
I think one of the main advantages of using the Hedera File Service is that updates to the file automatically take effect without needing to restart or upgrade the network, but this is not applicable to our case because we create the accounts on upgrade and perform a migration for the mirror nodes.
One more thing to consider is that removing a private key from the file will **not** uncreate the account, so basically we can add more private keys to the list, but not remove already created ones - or more precisely, we can remove already created ones, but that will only remove them from the file, not from the network!

Reply:
I should have worked out those implications, thanks!

Reply:
But, you know, in case someone _else_ doesn't work out those implications ... perhaps a section at the bottom of the doc for "other approaches considered" could have this. It would be a good place to remind readers that putting an address on this blocklist is not reversible!

Reply:
Added a section a the end of the doc for this.